tools/Makefile

```diff
@@ -16,10 +17,22 @@ OBJDUMP = $(or $(shell which objdump),xtensa-lx106-elf-objdump)
 SPIFFSFILES ?= $(patsubst $(FSSOURCE)%,%,$(shell find $(FSSOURCE) -name '*' '!' -name .gitignore ))
 
 #################################################################
-# Get the filesize of /bin/0x10000.bin
+# Get the filesize of /bin/0x10000.bin and SPIFFS sizing
 #
 
```

Comment:
On a default build with LFS enabled (set to 64k), the spiffs image gets built as `0x70000-*mb.img`. Yet, the actual file system location is at `0xb0000`.

I think this just reinforces that a partition table would be a very good thing...

Reply:
My concern with adding a partition table is that on the ESP8266, at least,  it would be just that: a bolt-on addition, and not properly integrated into the flash loader.  So how do you allocate partitions on the ESP8266.  Do we try to adopt the same conventions as on the ESP32, and if so how do we integrate them?

IMO, it would be better have one of two apporaches:
1.  You can define the base and size of both SPIFFS and LFS as config defines.  If you do then
    -  You can fix the absolute address of the LFS using the `-a` option, and ditto with spiffsimg
    -  `user_main.c` should do some bounds checks to make sure that everything fits properly
2.  You don't in which case _both_ fall back to using the platform flash allocator, and in this case there is nothing to stop the make process generate an internal table using the defines at (1) and the allocator uses these if set.   

Incidentally you can set the LFS size to any multiple of 4Kb up to 256Kb.

The botch with SPIFFS is that the format is _not_ PI, (though it could trivially be so).  If it were then the SPIFFS images would also be PI.  This would make life _**so**_ much simpler.

Reply:
I'd be happy enough to go with your approach, especially in the interest of getting LFS going. As long as the size/location gets set at compile time so we don't have any allocation races/ordering issues at runtime I think we're good.

As for spiffs, last I checked it's position independent - the phys/logic address translation takes place in the read/write accessor routines passed in to spiffs on initialisation. Not sure what makes you think it isn't position independent?

Reply:
> Not sure what makes you think it isn't position independent?

`spiffimg` requires you to specify an absolute location for the image.  I'll meed to have a look at the source to see why it does this, but if SPIFFS is PI then `spiffimg` should be as well.  This would make life easier for everyone.

Reply:
As the guy who wrote the initial `spiffsimg` before @pjsg co-opted into NodeMCU: nah, that's not what that argument really says. That's just one way to do the max fs size calculation :)
I readily confess that it should've all been documented better though...

Reply:
Ah, the penny drops re the difference between the `-c` and `-S`, so yes, it should have been better documented :grin:  In practice we have two use cases for a SPIFFS:
1.  The Lua app makes little use of SPIFFS, so it just needs to be "big enough"  (because of the flashing and startup overhead).  I typically use a 64Kb and work up from that is 64Kb multiples, simply because during development testing, it is often quicker to reflash the SPIFFS than use other ways to update it.
2/  The Lua developer isn't sure how much SPIFFS is going to used, so when in doubt then make it as big as possible.
I suspect that most developers start with (2) but if you reflash your firmware a lot then you end up doing (1).